### PR TITLE
docs: decimal type

### PR DESCRIPTION
**Pull request by QuenKar**
## What's Changed in this PR

*Describe the change in this PR*

decimal type docs

## Checklist

- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/v0.4/en/reference/data-types.md [hunk 1]
```diff
@@ -47,6 +47,44 @@ The following table lists the corresponding alias.
...
+## Decimal
+
+GreptimeDB supports `decimal` type, which is a fixed-point type. 
+
+It is represented as `decimal(precision, scale)`, where `precision` is the total number of digits and `scale` is the 
+number of digits in the fractional part. For example, `123.45` has a precision of 5 and a scale of 2.
+
+> ⚠️：It is an undefined behavior if we use `Decimal(3, 2)` to store `123.45`.
+
+- precision can range from [1, 38].
```
Comment (nicecui) [inline docs/v0.4/en/reference/data-types.md:59]:
```suggestion
- `precision` can range from [1, 38]. 
```

File: docs/v0.4/en/reference/data-types.md [hunk 2]
```diff
@@ -47,6 +47,44 @@ The following table lists the corresponding alias.
...
+
+GreptimeDB supports `decimal` type, which is a fixed-point type. 
+
+It is represented as `decimal(precision, scale)`, where `precision` is the total number of digits and `scale` is the 
+number of digits in the fractional part. For example, `123.45` has a precision of 5 and a scale of 2.
+
+> ⚠️：It is an undefined behavior if we use `Decimal(3, 2)` to store `123.45`.
+
+- precision can range from [1, 38]. 
+- scale can range from [0, precision].
```
Comment (nicecui) [inline docs/v0.4/en/reference/data-types.md:60]:
```suggestion
- `scale` can range from [0, precision].
```

File: docs/v0.4/en/reference/data-types.md [hunk 3]
```diff
@@ -47,6 +47,44 @@ The following table lists the corresponding alias.
...
 |TimestampNanosecond|Timestamp_ns , Timestamp(9)|
 
+## Decimal
+
+GreptimeDB supports `decimal` type, which is a fixed-point type. 
+
+It is represented as `decimal(precision, scale)`, where `precision` is the total number of digits and `scale` is the 
+number of digits in the fractional part. For example, `123.45` has a precision of 5 and a scale of 2.
+
+> ⚠️：It is an undefined behavior if we use `Decimal(3, 2)` to store `123.45`.
```
Comment (nicecui) [inline docs/v0.4/en/reference/data-types.md:57]: why do we emphasize this? If the inserted data does not comply with the decimal rules, an error should be thrown.

File: docs/v0.4/zh/reference/data-types.md [hunk 4]
```diff
@@ -43,6 +43,44 @@ variable-sized 类型的最大容量, 例如 `string` 和 `varbinary`，取决
...
+## Decimal
+
+GreptimeDB 支持 `decimal` 类型，这是一种定点类型。
+
+它表示为`decimal(precision, scale)`, `precision` 是数位的总数， `scale` 是小数部分的位数。
+例如, `123.45` 的 precision = 5 ， scale = 2.
+
+> ⚠️：用`Decimal(3, 2)` 存储 `123.45` 是未定义的行为
+
+- precision 的范围 [1, 38].
```
Comment (nicecui) [inline docs/v0.4/zh/reference/data-types.md:55]:
```suggestion
- `precision` 的范围 [1, 38].
```

File: docs/v0.4/zh/reference/data-types.md [hunk 5]
```diff
@@ -43,6 +43,44 @@ variable-sized 类型的最大容量, 例如 `string` 和 `varbinary`，取决
...
+- scale 的范围 [0, precision].
```
Comment (nicecui) [inline docs/v0.4/zh/reference/data-types.md:56]:
```suggestion
- `scale` 的范围 [0, precision].
```

File: docs/v0.4/zh/reference/data-types.md [hunk 6]
```diff
@@ -43,6 +43,43 @@ variable-sized 类型的最大容量, 例如 `string` 和 `varbinary`，取决
...
+GreptimeDB 支持 `decimal` 类型，这是一种定点类型。
+
+它表示为`decimal(precision, scale)`, `precision` 是数位的总数， `scale` 是小数部分的位数。
+例如, `123.45` 的 precision = 5 ， scale = 2.
+
+
+- `precision` 的范围 [1, 38].
+- `scale` 的范围 [0, precision].
+
+如果没有指定precision和scale，默认的decimal类型是 `decimal(38, 10)`.
```
Comment (xtang) [inline docs/v0.4/zh/reference/data-types.md:57]:
```suggestion
如果没有指定 precision 和 scale，默认的 decimal 类型是 `decimal(38, 10)`.
```

**Review by xtang — APPROVED** [review]

**Pull request merged**